### PR TITLE
[Evil] EAPI -> EVIL_API

### PR DESCRIPTION
**Pull request by joaoantoniocardoso**

### CODE REVIEW

**Comment by felipealmeida** [comment]
Tem como fazer um squash antes de mergear/rebase/cherry-pick ? Não faz sentido ficar com 2 commits

**Comment by joaoantoniocardoso** [comment]
> Tem como fazer um squash antes de mergear/rebase/cherry-pick ? Não faz sentido ficar com 2 commits

Sim, a gente tem usado **_squash and merge_** .

File: src/lib/evil/evil_api.h [hunk 1]
```diff
@@ -1,25 +1,25 @@
 #ifndef EVIL_API_H
 #define EVIL_API_H
 
-#ifdef EAPI
-#undef EAPI
+#ifdef EVIL_API
+#undef EVIL_API
```
Comment (Coquinho) [inline src/lib/evil/evil_api.h:5]: If `EVIL_API` is already defined shouldn't it yield an error? As it shouldn't be already defined?

**Comment by joaoantoniocardoso** [comment]
> Shouldn't this `evil_api.h` use `EVIL_STATIC` and `EVIL_BUILD`?

Yeah, you're right, I'll change it to work in the same way of the other APIs.

**Comment by joaoantoniocardoso** [comment]
Writing just to remember that #59 and #159 is adding new functions that this patch is not seeing yet.

File: src/lib/evil/evil_api.h [hunk 2]
```diff
@@ -1,26 +1,44 @@
 #ifndef EVIL_API_H
 #define EVIL_API_H
 
-#ifdef EAPI
-#undef EAPI
+#ifdef EVIL_API
+#error EVIL_API should not be already defined
 #endif
```
Comment (JPTIZ) [inline src/lib/evil/evil_api.h:6]: If we're using `evil_api.h`, there's no reason to check this at all actually, since the include-guards will handle it.

**Pull request merged**